plottr/data/datadict.py

```diff
@@ -362,15 +364,17 @@ def structure(self, add_shape: bool = False,
         :param same_type: if `True`, return type will be the one of the
                           object this is called on. Else, DataDictBase.
 
-        :return: The DataDictBase containing the structure only.
+        :return: The DataDict containing the structure only. The exact type
+                     is the same as the type of ``self``
+
         """
         if add_shape:
             warnings.warn("'add_shape' is deprecated and will be ignored",
                           DeprecationWarning)
         add_shape = False
 
         if self.validate():
-            s = DataDictBase()
+            s = self.__class__()
```

Comment:
this might not be correct because it could be that this method is supposed to create the "base" class and that can be "promoted" later if/as needed. @wpfff to help here :)

Reply:
Yes this is indeed a change of the code but it is unclear what is right solution. Both ``__add__`` and ``add_data`` of ``DataDict``` uses it without promoting but ``expand`` of ``DataDict``` does promote. It seems cleaner and less error prone to me to have it return the same type as self but happy to do it the other way around too

Reply:
let's go with the safe solution, unless @wpfff tells otherwise :)

Reply:
The reason this returned ``DataDictBase`` is that the structure, in general, might not be a valid higher type (it doesn't contain data). But i agree that it'd be cleaner if it was the same type. I don't see what it would break, so let's accept Jens' change.